src/interface/label.hpp

```diff
@@ -0,0 +1,60 @@
+//  SuperTux
+//  Copyright (C) 2020 A. Semphris <semphris@protonmail.com>
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef HEADER_SUPERTUX_INTERFACE_LABEL_HPP
+#define HEADER_SUPERTUX_INTERFACE_LABEL_HPP
+
+#include <SDL.h>
+
+#include "editor/widget.hpp"
+#include "video/drawing_context.hpp"
+
+class InterfaceLabel : public Widget
+{
+public:
+  InterfaceLabel();
+  InterfaceLabel(Rectf rect, std::string label);
+  virtual ~InterfaceLabel() {}
+
+  virtual void draw(DrawingContext& context) override;
+  virtual bool on_mouse_motion(const SDL_MouseMotionEvent& motion) override;
+
+  void set_rect(Rectf rect) { m_rect = rect; }
+  Rectf get_rect() { return m_rect; }
+
+  void set_label(std::string label) { m_label = label; }
+  std::string get_label() { return m_label; }
+
+  bool fits(std::string text);
+  std::string get_truncated_text();
```

Comment:
🖼️

Reply:
Both `fits`and `get_truncated_text`, yes :+1: